src/artifacts.js

```diff
@@ -727,7 +728,15 @@ api.declare({
       });
 
       if (headRes.headers['x-amz-meta-content-sha256'] !== artifact.details.contentSha256) {
-        throw new Error('S3 object does not have correct Content-Sha256 metadata');
+        let msg = [
+          'Expected S3 object to have SHA256 of ',
+          artifact.details.contentSha256,
+          ' but found it to have ',
+          headRes.headers['x-amz-meta-content-sha256'] || 'no value',
+          '',
+        ].join('"');
+        console.dir(headRes);
+        throw new Error(msg);
```

Comment:
Maybe remove `console.dir` would be nice..

The right thing is probably to report internal error to sentry if this happens, as it should be possible. But we should validate that we get a  `200` response before it's an internal error. I don't mind you not doing that, because this PR doesn't touch that :)